### DISCUSSION TITLE
Update all lock files in one pull request while the rest of dependencies by `parentDir`

### DISCUSSION THREAD

**Discussion by hongbo-miao** [renovatebot/renovate#22230] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Wanted end result.

I have a mono repo. It has many lock files can update.

Here is my **renovate.json5**:

```json5
{
  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
  extends: [
    'config:base',
  ],
  lockFileMaintenance: {
    enabled: true,
    commitMessageExtra: '({{packageFile}})',
  },
  additionalBranchPrefix: '{{parentDir}}-',
  semanticCommitScope: '{{#if parentDir}}{{parentDir}}{{else}}deps{{/if}}',
  prHourlyLimit: 0,
}
```

Sometimes, some week, it updates normally. In other weeks, it only updates three (I guess maybe because of some rate limit, but I do have `prHourlyLimit: 0` 🤔)

![image](https://github.com/renovatebot/renovate/assets/3375461/63496742-cd36-4b82-9f98-c05436a79d17)

Actually I am hoping to update all lock files in one pull request. But for the rest of dependencies, I still hope them update by `parentDir`.

Is there a way? Thanks! 😃

### What you tried so far.

I checked https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

But didn't find the related info.

**Comment by viceice** [comment, marked as answer]
move `additionalBranchPrefix` and `semanticCommitScope` to a package rule with `matchUpdateTypes`[^1]

[^1]: https://docs.renovatebot.com/configuration-options/#matchupdatetypes

**Reply by hongbo-miao** [reply]
Thanks @viceice , it works now! 😃

Here is the updated **renovate.json5**:

```json5
{
  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
  extends: [
    'config:base',
  ],
  lockFileMaintenance: {
    enabled: true,
  },
  packageRules: [
    {
      matchUpdateTypes: [
        'major',
        'minor',
        'patch',
      ],
      additionalBranchPrefix: '{{parentDir}}-',
      semanticCommitScope: '{{#if parentDir}}{{parentDir}}{{else}}deps{{/if}}',
    },
  ],
  prHourlyLimit: 0,
}
```
